node_modules/col-core/lib/db.js

```diff
@@ -78,19 +78,323 @@ var init = module.exports.init = function(callback) {
+  var objectTypes = ['canvas_submission', 'canvas_discussion', 'asset', 'comment'];
+
+  /**
+   * The `activity` table will keep track of actions that users make in the system. This can
+   * include things such as:
+   *  - liking or disliking assets
+   *  - commenting on assets
+   *  - submitting an assignment
+   *  - posting to a Canvas discussion
+   *  - creating a board
+   *  - ...
+   *
+   * As an activity can be related to different things, it's hard to enforce referential integrity without
+   * adding lots of unused foreign keys to the table. That's why a more general approach has been taken:
+   *  - `object_id` will be an identifier (e.g., canvas assignment id)
+   *  - `object_type` will hold information about the type the id points to (e.g., a canvas assignment)
+   *
+   * @property  {String}        type                The type of the activity
+   * @property  {Number}        object_id           The id of the object that was involved in the activity. This can be a Canvas assignment id, a comment id, an asset id, etc
+   * @property  {String}        object_type         The type of object that was involved in the activity
+   */
+  var Activity = module.exports.Activity = sequelize.define('activity', {
+    'type': Sequelize.ENUM(activityTypes),
+    'object_id': Sequelize.INTEGER,
+    'object_type': Sequelize.ENUM(objectTypes)
+  }, {
+    'underscored': true
+  });
+
+  // Each activity happens within a course
+  Activity.belongsTo(Course, {'onDelete': 'CASCADE'});
+
+  // The user who triggered the activity
+  Activity.belongsTo(User, {'as': 'actor', 'onDelete': 'CASCADE'});
+
+  // The target is the user who was indirectly involved in the activity. For example,
+  // when John likes a comment from Bella, John will be the actor and Bella will be the target.
+  // When users drop out of a course they will be removed from the system. That doesn't
+  // mean however that people who liked or commented on assets generated by those users
+  // should lose points in their engagement index. That's why target ids will be set to
+  // null when such a user is removed from the system
+  Activity.belongsTo(User, {'as': 'target', 'onDelete': 'SET NULL'});
+
+  /**
+   * The `activity_type` table will keep track of the different activities that can occur
+   * and the weights that are associated with them.
+   *
+   * @property  {String}        type                The type of activity. For example `like`, `comment`, ..
+   * @property  {String}        name                The human-readable name of an activity. For example `Like`, `Comment`, ..
+   * @property  {Number}        points              The number of points this activity adds towards a user's engagement index
+   * @property  {Boolean}       enabled             Whether activities of this type should be added towards a user's engagement index
+   */
+  var ActivityType = module.exports.ActivityType = sequelize.define('activity_type', {
+    'type': {
```

Comment:
We could also hold a default points configuration object in the codebase and only store the overrides in this table. In that case, the display name could probably just be part of the default configuration and we wouldn't have to store it in the database.